spec/models/assignment_spec.rb

```diff
@@ -316,4 +316,31 @@ def generated_csv(t_assignment, t_options)
       expect(generated_csv(assignment, @options)).to eq(expected_csv)
     end
   end
+
+  #describe '#check_directory_path' do
+  #  create(:assignment)
+  #  @assignment = create(:assignment)
+  #  expect(check_directory_path).to eq(false)
+  #end
+
+  describe 'check_directory_path' do
+    it 'returns false if directory path already in use' do
+      @assignment = create(:assignment)
```

Comment:
For unit test, I think you do not need to create objects. You could `build` objects instead.